spec.html

```diff
@@ -24817,55 +25337,79 @@ <h1>
-          1. If _element_ is an abrupt completion, then
+          1. If IsStatic of _e_ is *false*, let _result_ be Completion(ClassElementEvaluation of _e_ with argument _proto_).
+          1. Else, let _result_ be Completion(ClassElementEvaluation of _e_ with argument _F_).
+          1. If _result_ is an abrupt completion, then
             1. Set the running execution context's LexicalEnvironment to _env_.
             1. Set the running execution context's PrivateEnvironment to _outerPrivateEnvironment_.
-            1. Return ? _element_.
-          1. Set _element_ to _element_.[[Value]].
-          1. If _element_ is a PrivateElement, then
-            1. Assert: _element_.[[Kind]] is either ~method~ or ~accessor~.
-            1. If IsStatic of _e_ is *false*, let _container_ be _instancePrivateMethods_.
-            1. Else, let _container_ be _staticPrivateMethods_.
-            1. If _container_ contains a PrivateElement whose [[Key]] is _element_.[[Key]], then
-              1. Let _existing_ be that PrivateElement.
-              1. Assert: _element_.[[Kind]] and _existing_.[[Kind]] are both ~accessor~.
-              1. If _element_.[[Get]] is *undefined*, then
-                1. Let _combined_ be PrivateElement { [[Key]]: _element_.[[Key]], [[Kind]]: ~accessor~, [[Get]]: _existing_.[[Get]], [[Set]]: _element_.[[Set]] }.
-              1. Else,
-                1. Let _combined_ be PrivateElement { [[Key]]: _element_.[[Key]], [[Kind]]: ~accessor~, [[Get]]: _element_.[[Get]], [[Set]]: _existing_.[[Set]] }.
-              1. Replace _existing_ in _container_ with _combined_.
-            1. Else,
-              1. Append _element_ to _container_.
-          1. Else if _element_ is a ClassFieldDefinition Record, then
-            1. If IsStatic of _e_ is *false*, append _element_ to _instanceFields_.
+            1. Return ? _result_.
+          1. Let _element_ be _result_.[[Value]].
+          1. If _element_ is a ClassElementDefinition Record, then
+            1. If IsStatic of _e_ is *false*, append _element_ to _instanceElements_.
             1. Else, append _element_ to _staticElements_.
-          1. Else if _element_ is a ClassStaticBlockDefinition Record, then
+          1. Else,
+            1. Assert: _element_ is a ClassStaticBlockDefinition Record.
             1. Append _element_ to _staticElements_.
         1. Set the running execution context's LexicalEnvironment to _env_.
-        1. If _classBinding_ is not *undefined*, then
-          1. Perform ! _classEnv_.InitializeBinding(_classBinding_, _F_).
-        1. Set _F_.[[PrivateMethods]] to _instancePrivateMethods_.
-        1. Set _F_.[[Fields]] to _instanceFields_.
+        1. Let _instanceExtraInitializers_ be a new empty List.
+        1. Let _staticExtraInitializers_ be a new empty List.
+        1. For each element _e_ of _staticElements_, do
+          1. If _e_.[[Kind]] is not ~field~, then
```

Comment:
This assumes that _e_ is a ClassElementDefinition Record; however, it could also be a ClassStaticBlockDefinition Record (see a few lines above). So this should probably say sth like:
> If _e_ is a ClassElementDefinition Record and _e_.[[Kind]] is not field, then